worlds/stardew_valley/mods/logic/deepwoods.py

```diff
@@ -8,34 +7,35 @@
 from ... import options
 
 
-def can_reach_woods_depth(self, depth: int) -> StardewRule:
+def can_reach_woods_depth(vanilla_logic, depth: int) -> StardewRule:
     tier = int(depth / 25) + 1
     rules = []
     if depth > 10:
-        rules.append(self.has(Craftable.bomb) | self.has_tool(Tool.axe, ToolMaterial.iridium))
+        rules.append(vanilla_logic.has(Craftable.bomb) | vanilla_logic.has_tool(Tool.axe, ToolMaterial.iridium))
     if depth > 30:
-        rules.append(self.received(ModTransportation.woods_obelisk))
+        rules.append(vanilla_logic.received(ModTransportation.woods_obelisk))
     if depth > 50:
-        rules.append(self.can_do_combat_at_level(Performance.great) & self.can_cook())
-    if self.options[options.SkillProgression] == options.SkillProgression.option_progressive:
+        rules.append(vanilla_logic.can_do_combat_at_level(Performance.great) & vanilla_logic.can_cook() &
+                     vanilla_logic.received(ModTransportation.woods_obelisk))
+    if vanilla_logic.options[options.SkillProgression] == options.SkillProgression.option_progressive:
         combat_tier = min(10, max(0, tier + 5))
-        rules.append(self.has_skill_level(Skill.combat, combat_tier))
+        rules.append(vanilla_logic.has_skill_level(Skill.combat, combat_tier))
     return And(rules)
 
 
-def has_woods_rune_to_depth(self, floor: int) -> StardewRule:
-    if (self.options[options.TheMinesElevatorsProgression] ==
+def has_woods_rune_to_depth(vanilla_logic, floor: int) -> StardewRule:
+    if (vanilla_logic.options[options.TheMinesElevatorsProgression] ==
             options.TheMinesElevatorsProgression.option_progressive or
-            self.options[options.TheMinesElevatorsProgression] ==
+            vanilla_logic.options[options.TheMinesElevatorsProgression] ==
             options.TheMinesElevatorsProgression.option_progressive_from_previous_floor):
-        return self.received("Progressive Wood Obelisk Sigils", count=int(floor / 10))
+        return vanilla_logic.received("Progressive Wood Obelisk Sigils", count=int(floor / 10))
     return True_()
 
 
-def can_chop_to_depth(self, floor: int) -> StardewRule:
+def can_chop_to_depth(vanilla_logic, floor: int) -> StardewRule:
     previous_elevator = max(floor - 10, 0)
     previous_previous_elevator = max(floor - 20, 0)
-    return ((has_woods_rune_to_depth(self, previous_elevator) &
-             can_reach_woods_depth(self, previous_elevator)) |
-            (has_woods_rune_to_depth(self, previous_previous_elevator) &
-             can_reach_woods_depth(self, previous_previous_elevator)))
+    return ((has_woods_rune_to_depth(vanilla_logic, previous_elevator) &
```

Comment:
I'm not sure I understand the logic behind this.
Either [Can reach depth - 10 and has rune for it] or [can reach depth - 10 and has rune for it]. Isn't it redundant? If you can reach depth - 10, you can obviously reach depth - 20, right? Is there a typo somewhere, or am I misunderstanding it?